app/controllers/application_controller.rb

```diff
@@ -33,6 +33,14 @@ class ApplicationController < ActionController::Base # rubocop:disable Metrics/C
 
   skip_before_action :handle_two_factor_authentication
 
+  # BAD BAD BAD
+  before_action :log_action
```

Comment:
What if we guarded this whole block on `ENV['COVERAGE']`?

```ruby
if ENV['COVERAGE']
  before_action :log_actoin
  def log_action
    # ...
  end
end
```

- I'd switch the `puts` to a `Rails.logger.info`
- Just logging the action name doesn't tell us what view was rendered, we sometimes call `render :template_name` explicitly with a different template

I know that we configured the Rails logger to be more terse/JSON-y but the default Rails logger prints out every template and partial that gets rendered, so maybe we should just switch loggers if `ENV['COVERAGE']` is set?

Reply:
I mean, for one thing I don't think it belongs in the actual application_controller.

The idea was to get some sort of list to compare against actual views to see how accessibility coverage looks, I'm not sure sprinkling them throughout the log helps.  Maybe there's some way better way?  This felt a little like a 1-off for the "aXe Audit" task.

🤔 Maybe a better approach is to figure out a way to scan `test.log` and pull out any controller/views or whatever listed there.

Reply:
Yup! With a little diff like this to disable lograge in tests...

```diff
diff --git a/config/environments/test.rb b/config/environments/test.rb
index 008648891..4cbf9be08 100644
--- a/config/environments/test.rb
+++ b/config/environments/test.rb
@@ -28,7 +28,7 @@ Rails.application.configure do
   config.assets.digest = ENV.key?('RAILS_DISABLE_ASSET_DIGEST') ? false : true
 
   config.middleware.use RackSessionAccess::Middleware
-  config.lograge.enabled = true
+  config.lograge.enabled = false
```

and a little grep on the log...

```
identity-idp:master> tail -f log/test.log | grep Render
I, [2020-03-09T09:29:03.033015 #43919]  INFO -- :   Rendering sign_up/registrations/new.html.erb within layouts/application
I, [2020-03-09T09:29:04.322861 #43919]  INFO -- :   Rendered shared/_sp_alert.html.slim (12.4ms)
I, [2020-03-09T09:29:04.393012 #43919]  INFO -- :   Rendered shared/_recaptcha.html.slim (4.9ms)
I, [2020-03-09T09:29:04.398652 #43919]  INFO -- :   Rendered shared/_cancel.html.slim (3.0ms)
I, [2020-03-09T09:29:04.404321 #43919]  INFO -- :   Rendered sign_up/registrations/new.html.erb within layouts/application (1370.9ms)
I, [2020-03-09T09:29:04.407942 #43919]  INFO -- :   Rendering layouts/base.html.slim
I, [2020-03-09T09:29:09.312313 #43919]  INFO -- :   Rendered shared/_usa_banner.html.slim (2.6ms)
I, [2020-03-09T09:29:09.317749 #43919]  INFO -- :   Rendered shared/_nav_lite.html.slim (3.0ms)
I, [2020-03-09T09:29:09.325338 #43919]  INFO -- :   Rendered shared/_flashes.html.slim (4.3ms)
```

Reply:
yeah, way better idea.  Closing this PR !